exp/addons/controllers/clusterresourceset_controller.go

```diff
@@ -69,40 +69,29 @@ type ClusterResourceSetReconciler struct {
 }
 
 func (r *ClusterResourceSetReconciler) SetupWithManager(mgr ctrl.Manager, options controller.Options) error {
-	controller, err := ctrl.NewControllerManagedBy(mgr).
+	err := ctrl.NewControllerManagedBy(mgr).
 		For(&addonsv1.ClusterResourceSet{}).
 		Watches(
 			&source.Kind{Type: &clusterv1.Cluster{}},
 			&handler.EnqueueRequestsFromMapFunc{ToRequests: handler.ToRequestsFunc(r.clusterToClusterResourceSet)},
 		).
+		Watches(&source.Kind{Type: builder.OnlyMetadata(&corev1.ConfigMap{})},
+			&handler.EnqueueRequestsFromMapFunc{
+				ToRequests: handler.ToRequestsFunc(r.resourceToClusterResourceSet),
+			},
+		).
+		Watches(&source.Kind{Type: builder.OnlyMetadata(&corev1.Secret{})},
+			&handler.EnqueueRequestsFromMapFunc{
+				ToRequests: handler.ToRequestsFunc(r.resourceToClusterResourceSet),
+			},
+		).
 		WithOptions(options).
 		WithEventFilter(predicates.ResourceNotPaused(r.Log)).
-		Build(r)
+		Complete(r)
 	if err != nil {
 		return errors.Wrap(err, "failed setting up with a controller manager")
 	}
 
-	err = controller.Watch(
-		&source.Kind{Type: &corev1.ConfigMap{}},
-		&handler.EnqueueRequestsFromMapFunc{
-			ToRequests: handler.ToRequestsFunc(r.resourceToClusterResourceSet),
-		},
-		resourcepredicates.ResourceCreate(r.Log),
```

Comment:
If I'm not wrong this line went lost in the refactoring. Is it a problem?

Reply:
These predicates were mostly added to avoid reconciling too often, I don't see it as a huge problem for v0.3.x, we can add these back in v0.4.x